src-wasm/src/midi.rs

```diff
@@ -1,19 +1,34 @@
-use std::future::Future;
-use std::pin::Pin;
+use crate::EventHandler;
+use saturn_backend::midi::instruments::to_instrument;
+use saturn_backend::midi::note::MidiNote;
 use saturn_backend::syscall::{MidiHandler, MidiRequest};
+use std::pin::Pin;
+use std::{future::Future, sync::Arc};
 
-pub struct WasmMidi { }
+pub struct WasmMidi {
+    pub events: Arc<EventHandler>,
+}
 
 impl MidiHandler for WasmMidi {
-    fn play(&mut self, _request: &MidiRequest, _sync: bool) {
-        todo!()
+    fn play(&mut self, request: &MidiRequest, _sync: bool) {
+        let Some(name) = to_instrument(request.instrument as usize) else {
+            return;
+        };
+        self.events.send_midi_play(MidiNote {
+            sync: _sync,
+            name: name.to_owned(),
+            instrument: request.instrument as u64,
+            note: request.pitch as u64,
+            duration: request.duration as f64 / 1000f64,
+            volume: request.volume as u64,
+        })
     }
 
-    fn install(&mut self, _instrument: u32) -> Pin<Box<dyn Future<Output=bool> + Send>> {
-        todo!()
+    fn install(&mut self, _instrument: u32) -> Pin<Box<dyn Future<Output = bool> + Send>> {
```

Comment:
I forget why I didn't just use `async_trait` for this, but I remember the reason being very complicated. Sorry for the whole work-around.